hypertrace-trace-enricher/trace-reader/src/test/java/org/hypertrace/trace/reader/attributes/DefaultValueResolverTest.java

```diff
@@ -176,6 +179,77 @@ void resolvesExpressionProjections() {
             .blockingGet());
   }
 
+  @Test
+  void resolvesExpressionProjectionsForStringEqualsAndConditional() {
+    AttributeMetadata projectionMetadata =
+        AttributeMetadata.newBuilder()
+            .setDefinition(
+                AttributeDefinition.newBuilder()
+                    .setProjection(
+                        Projection.newBuilder()
+                            .setExpression(
+                                ProjectionExpression.newBuilder()
+                                    .setOperator(ProjectionOperator.PROJECTION_OPERATOR_CONDITIONAL)
+                                    .addArguments(
+                                        Projection.newBuilder().setExpression(
+                                            ProjectionExpression.newBuilder()
+                                                .setOperator(ProjectionOperator.PROJECTION_OPERATOR_STRING_EQUALS)
+                                                .addArguments(Projection.newBuilder().setAttributeId("TEST_SCOPE.second"))
+                                                .addArguments(Projection.newBuilder().setLiteral(LiteralValue.newBuilder().setStringValue("foo")))
+                                        )
+                                    )
+                                    .addArguments(
+                                        Projection.newBuilder().setAttributeId("TEST_SCOPE.first"))
+                                    .addArguments(
+                                        Projection.newBuilder()
+                                            .setLiteral(LiteralValue.newBuilder().setStringValue("null")))
```

Comment:
@aaron-steinfeld is this what you meant?

Reply:
Yes, but this is verifying the behavior I was afraid of. If I try to use the attribute reader to get a value and it returns a string literal of "null", the burden is on every consumer to differentiate that from a valid string. For example, in the entity reader, we'd try to create an entity with the id "null" if we don't look for this special string, which is based on pinot's conception of null.

The real behavior we want IMO is one of these:
- For the resolved single to be an error (the current api supports this, but it means the actual projection would need to throw which it doesn't currently)
- For the resolved single to be an unset literal value or a currently non-existent null-typed literal
- Change the attribute reader api to return a Maybe (this is a slight tweak of the first variation above, still requires the actual projection code to indicate an invalid result)